### PR TITLE
[HttpClient-changes] handler the request-response in vertx4

### PR DESCRIPTION
**Pull request by jcarranzan**
Motivation:

Context
Now in vertx 4.0.0 when you do some HttpClientRequest and get the result if you want to do after some operation (compose, succeed,etc) with the result before sending it, you have to add the response method that returns the Future

This change is shown in vertx examples in the core/http/proxy/client.java :
request.response().compose

vert-x3/vertx-examples@1ddfd8c#diff-330da0139766c23d1b970aa604bae5ed935b15ce6800348c696b0a5f6744f74b

### CODE REVIEW

File: asciidoc/modules/con_mg_migrating-applications-to-vertx-http-client.adoc [hunk 1]
```diff
@@ -201,3 +201,36 @@ response.end(ar -> {
   }
 });
 ----
++
+
+On the other hand, we can handle the response with some operations if we want to. You can see the difference in the next examples.
```
Comment (vietj) [inline asciidoc/modules/con_mg_migrating-applications-to-vertx-http-client.adoc:206]: instead of we use "you"

Comment (sangeetaraghu) [reply asciidoc/modules/con_mg_migrating-applications-to-vertx-http-client.adoc:206]: @jcarranzan @vietj Can we rephrase this to:

You can also handle the response with operations. The following examples demonstrate handling responses with operations.

Comment (jcarranzan) [reply asciidoc/modules/con_mg_migrating-applications-to-vertx-http-client.adoc]: ok I've updated it.
What do you think?

Thank you

File: asciidoc/modules/con_mg_migrating-applications-to-vertx-http-client.adoc [hunk 2]
```diff
@@ -203,7 +203,8 @@ response.end(ar -> {
 ----
 +
 
-On the other hand, we can handle the response with some operations if we want to. You can see the difference in the next examples.
+On the other hand, you can also handle the response with operations like onSuccess, compose, bodyHandler, etc.
+The following examples demonstrate handling responses with the operation onSuccess :
```
Comment (sangeetaraghu) [inline asciidoc/modules/con_mg_migrating-applications-to-vertx-http-client.adoc:208]: You can also handle the response with methods such as, `onSucces()`, `compose()`, and `bodyHandler()`.
The following examples demonstrate handling responses using the `onSuccess()` method.

File: asciidoc/modules/con_mg_migrating-applications-to-vertx-http-client.adoc [hunk 3]
```diff
@@ -203,7 +203,8 @@ response.end(ar -> {
...
 
 The following example shows an example using HTTP client in {VertX} {v3x} releases:
```
Comment (sangeetaraghu) [inline asciidoc/modules/con_mg_migrating-applications-to-vertx-http-client.adoc:210]: @jcarranzan Sorry missed these:

The following example shows how to use HTTP client in {VertX} {v3x} releases.

The following example shows how to use HTTP client with the `result()` method in {VertX} {v4}.

Can you please change the comment in both Vert.x 3 and 4 examples from:
//Handler some operation with the response
to 
//Code to handle HTTP response

File: asciidoc/modules/con_mg_migrating-applications-to-vertx-http-client.adoc (hunk 2 above)
Comment (jcarranzan) [reply asciidoc/modules/con_mg_migrating-applications-to-vertx-http-client.adoc:208]: I would like to point here, that is not only those 3 methods, there are a few methods more.
